astronomer/providers/google/cloud/sensors/gcs.py

```diff
@@ -78,3 +89,79 @@ def execute_complete(self, context, event=None):  # pylint: disable=unused-argum
             raise AirflowException(event["message"])
         self.log.info("File %s was found in bucket %s.", self.object, self.bucket)
         return event["message"]
+
+
+class GCSObjectsWithPrefixExistenceSensorAsync(GCSObjectsWithPrefixExistenceSensor):
+    def __init__(
+        self,
+        polling_interval: float = 5.0,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.polling_interval = polling_interval
+
+    def execute(self, context: "Context") -> None:
+        self.defer(
+            timeout=self.execution_timeout,
+            trigger=GCSPrefixBlobTrigger(
+                bucket=self.bucket,
+                prefix=self.prefix,
+                polling_period_seconds=self.polling_interval,
+                google_cloud_conn_id=self.google_cloud_conn_id,
+                hook_params=dict(delegate_to=self.delegate_to, impersonation_chain=self.impersonation_chain),
+            ),
+            method_name="execute_complete",
+        )
+
+    def execute_complete(
+        self, context: "Context", event: Optional[Dict[Any, Any]] = None
+    ) -> List[str]:  # pylint: disable=unused-argument
+        """
+        Callback for when the trigger fires - returns immediately.
+        Relies on trigger to throw an exception, otherwise it assumes execution was
+        successful.
+        """
+        self.log.info('Sensor checks existence of objects: %s, %s', self.bucket, self.prefix)
+        if event["status"] == "success":
+            return event["matches"]
+        raise AirflowException(event["message"])
+
+
+class GCSUploadSessionCompleteSensorAsync(GCSUploadSessionCompleteSensor):
+    def __init__(
```

Comment:
Docstring and comment missing explaining the sensor.

Reply:
it is inherited from sync version added doc string for only async inherited from base operator